dlib/CMakeLists.txt

```diff
@@ -1,1039 +1,391 @@
+option(DLIB_PNG_SUPPORT          "Enable PNG support"             ON)
+cmake_dependent_option(DLIB_USE_BLAS            "Enable BLAS support"      ON  "MKL_FOUND OR BLAS_FOUND" OFF)
+cmake_dependent_option(DLIB_USE_LAPACK          "Enable LAPACK support"    ON  "MKL_FOUND OR LAPACK_FOUND" OFF)
+cmake_dependent_option(DLIB_USE_AVX             "Enable AVX SIMD"          ON  "AVX_FOUND"         OFF)
+cmake_dependent_option(DLIB_USE_SSE             "Enable SSE SIMD"          ON  "SSE_FOUND"         OFF)
+cmake_dependent_option(DLIB_USE_NEON            "Enable NEON SIMD"         ON  "NEON_FOUND"        OFF)
+cmake_dependent_option(DLIB_GIF_SUPPORT         "Enable GIF support"       ON  "GIF_FOUND"         OFF)
+cmake_dependent_option(DLIB_WEBP_SUPPORT        "Enable WEBP support"      ON  "WEBP_FOUND"        OFF)
+cmake_dependent_option(DLIB_NO_GUI_SUPPORT      "Disable GUI support"      OFF "X11_FOUND OR WIN32" ON)
+cmake_dependent_option(DLIB_LINK_WITH_SQLITE3   "Enable Sqlite3 support"   ON  "SQLite3_FOUND"     OFF)
+cmake_dependent_option(DLIB_USE_FFMPEG          "Enable FFMPEG support"    ON  "FFMPEG_FOUND"      OFF)
+cmake_dependent_option(DLIB_USE_CUDA            "Enable CUDA support"      ON  "CUDAToolkit_FOUND" OFF)
+cmake_dependent_option(DLIB_USE_MKL_SEQUENTIAL  "Enable MKL sequential"    ON  "MKL_FOUND"         OFF)
+cmake_dependent_option(DLIB_USE_MKL_WITH_TBB    "Enable MKL tbb"           ON  "MKL_FOUND"         OFF)
+cmake_dependent_option(DLIB_USE_MKL_FFT         "Enable MKL FFT"           ON  "MKL_FOUND"         OFF)
+
+# dlib
+add_library(dlib
+   base64/base64_kernel_1.cpp
+   bigint/bigint_kernel_1.cpp
+   bigint/bigint_kernel_2.cpp
+   bit_stream/bit_stream_kernel_1.cpp
+   entropy_decoder/entropy_decoder_kernel_1.cpp
+   entropy_decoder/entropy_decoder_kernel_2.cpp
+   entropy_encoder/entropy_encoder_kernel_1.cpp
+   entropy_encoder/entropy_encoder_kernel_2.cpp
+   md5/md5_kernel_1.cpp
+   tokenizer/tokenizer_kernel_1.cpp
+   unicode/unicode.cpp
+   test_for_odr_violations.cpp
+   fft/fft.cpp)
 
-include(cmake_utils/set_compiler_specific_options.cmake)
-
-
-# Adhere to GNU filesystem layout conventions
-include(GNUInstallDirs)
-
-if (POLICY CMP0075)
-   cmake_policy(SET CMP0075 NEW)
-endif()
-
-# default to a Release build (except if CMAKE_BUILD_TYPE is set)
-include(cmake_utils/release_build_by_default)
-
-# Set DLIB_VERSION in the including CMake file so they can use it to do whatever they want. 
-get_directory_property(has_parent PARENT_DIRECTORY)
-if(has_parent)
-   set(DLIB_VERSION ${VERSION} PARENT_SCOPE)
-   if (NOT DEFINED DLIB_IN_PROJECT_BUILD)
-      set(DLIB_IN_PROJECT_BUILD true)
+add_library(dlib::dlib ALIAS dlib)
+target_include_directories(dlib
+   INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/..>
+   INTERFACE $<INSTALL_INTERFACE:include>)
+target_compile_options(dlib PUBLIC $<${IS_CLANG_OLD}:-ftemplate-depth=500>)
+target_compile_options(dlib PUBLIC $<${IS_WIN32}:/bigobj>)
```

Comment:
/bigobj needs to be gated to MSVC.  Someone using some other random compiler on windows would get an error since /bigobj is a MSVC option.  People use mingw sometimes still I think. I periodically get emails from people using other compilers on windows doing various things.